tests/vision/checks/model_evaluation/single_dataset_performance_test.py

```diff
@@ -0,0 +1,126 @@
+
+    # Assert
+    assert_that(result.value.Value.mean(), close_to(0.416, 0.001))
+
+
+def test_detection_w_params(coco_train_visiondata, mock_trained_yolov5_object_detection, device):
+    # params that should run normally
+    check = SingleDatasetPerformance(alternative_scorers={'f1': ObjectDetectionTpFpFn(evaluting_function='f1')})
+    result = check.run(coco_train_visiondata, mock_trained_yolov5_object_detection, device=device)
+    assert_that(result.value.Value.mean(), close_to(0.505, 0.001))
+
+
+def test_classification_defaults(mnist_dataset_train, mock_trained_mnist, device):
+    # Arrange
+    check = SingleDatasetPerformance()
+
+    # Act
+    result = check.run(mnist_dataset_train, mock_trained_mnist, device=device)
+
+    # Assert
+    assert_that(result.value.Value.mean(), close_to(0.98, 0.001))
+
+
+def test_classification_custom_scorer(mnist_dataset_test, mock_trained_mnist, device):
+    # Arrange
+    check = SingleDatasetPerformance(alternative_scorers={'kappa': CustomScorer(cohen_kappa_score)})
+
+    # Act
+    result = check.run(mnist_dataset_test, mock_trained_mnist, device=device, n_samples=None)
+
+    # Assert
+    assert_that(result.value.Value.mean(), close_to(0.979, 0.001))
+
+
+def test_condition_greater_than(mnist_dataset_test, mock_trained_mnist, device):
+    check = SingleDatasetPerformance().add_condition_greater_than(0.8)\
+        .add_condition_greater_than(1.0, ['Precision'])\
+        .add_condition_greater_than(0.5, ['Accuracy'])\
+        .add_condition_greater_than(0.5, class_mode='a')
+
+    # Act
+    result = check.run(mnist_dataset_test, mock_trained_mnist, device=device)
+
+    # Assert
+    assert_that(result.conditions_results[0], equal_condition_result(
+        is_pass=True,
+        name='Score is greater than 0.8',
+        details='Passed for all of the mertics.'
+        ))
+
+    assert_that(result.conditions_results[1], equal_condition_result(
+        is_pass=False,
+        name='Score is greater than 1.0',
+        details='Failed for metrics: [\'Precision\']'
+        ))
+
+    assert_that(result.conditions_results[2], equal_condition_result(
+        is_pass=False,
+        category=ConditionCategory.ERROR,
+        details='Exception in condition: DeepchecksValueError: The requested metric was not calculated, the metrics '
+                'calculated in this check are: [\'Precision\' \'Recall\'].',
+        name='Score is greater than 0.5'
+        ))
+
+    assert_that(result.conditions_results[3], equal_condition_result(
+        is_pass=False,
+        category=ConditionCategory.ERROR,
+        details='Exception in condition: DeepchecksValueError: class_mode expected be one of the classes in the check '
+                'results or any or all, recieved a.',
+        name='Score is greater than 0.5'
+        ))
+
+def test_reduce_output(mnist_dataset_test, mock_trained_mnist, device):
```

Comment:
Would add an additional test that receives scorers a list of str

Reply:
@Nadav-Barak this is managed by the function get scorers dict, and therefore the test belongs to the metrics module tests and not here